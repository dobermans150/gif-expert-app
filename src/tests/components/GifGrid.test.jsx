import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas GifGrid', () => {
  test('Debe de retornar un gif grid con sus items necesarios', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category="one punch" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
    const gifs = [
      {
        id: 'abc',
        url: 'https://localhost/cualquier/cosa.png',
        title: 'cualquiercosa',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category="one punch" />);

    /* expect(wrapper).toMatchSnapshot(); */
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
