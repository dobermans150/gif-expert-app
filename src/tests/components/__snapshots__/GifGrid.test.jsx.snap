// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifGrid.jsx Debe  de retornar un component  1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="one punch GIF by Adult Swim"
    src="https://media1.giphy.com/media/l3vRlt8kty8KKeHni/giphy.gif?cid=12100ec9lzk6gum9djjurk3i209s81k9ktrp8luz4haam66q&rid=giphy.gif&ct=g"
  />
  <p>
    one punch GIF by Adult Swim
  </p>
</div>
`;
